### PR TITLE
Adding Spinner and Dynamic Router

### PR DESCRIPTION
**Pull request by Mus90**
I change the loading to be false when catching an error , so the spinner will not pop up

### CODE REVIEW

**Comment by heshamelmasry77** [comment]
> _No description provided._

when you name a component don't use` ,` please.

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
+        console.log("res:", res);
+        this.setState({ persons: res.data, loading: false }, () => {
+          console.log("state: ", this.state.persons);
         });
-      }).catch((error) => {
-      this.setState({error: true});
-    })
+      })
+      .catch((error) => {
+        this.setState({ error: true });
```
Comment (heshamelmasry77) [inline src/App.js]: what if we catch an error the loader will keep true and showing, you must also set state loader to false here if we have an error

File: src/App.js [hunk 2]
```diff
@@ -28,11 +33,12 @@ export default class App extends Component {
     return (
       <div className="App">
         <ul>
-          {
-            !this.state.error && persons.map(person =>
-              <PersonView key={person.id} personData={person}/>
-            )
-          }
+          {this.state.loading && <Loader />}
```
Comment (heshamelmasry77) [inline src/App.js]: move this line of code after line 43, usually the loader must be at the end of the page.

File: src/App.js [hunk 3]
```diff
@@ -1,25 +1,30 @@
-import React, {Component} from "react";
+import React, { Component } from "react";
 import axios from "axios";
-import './App.scss';
+import "./App.scss";
 import PersonView from "./PersonView";
-
+import Loader from "react-loader-spinner";
```
Comment (heshamelmasry77) [inline src/App.js]: please import al the libraries first then import the style sheets.
```
import React, { Component } from "react";
import axios from "axios";
import Loader from "react-loader-spinner";
```
then 
style sheets here.

File: src/PersonView.js [hunk 4]
```diff
@@ -1,57 +1,32 @@
...
-        <h3>Address</h3>
-        <ul>
-          <li>{person.address.city}</li>
-          <h3>link to the single page details /user/{person.id}</h3>
-        </ul>
-      </div>
-      }
-    </li>
-  </div>;
+  return (
```
Comment (heshamelmasry77) [inline src/PersonView.js]: adding `()` is always a good practise.

**Review by heshamelmasry77 — APPROVED** [review]
good but also take a look on my other comments

**Comment by Mus90** [comment]
I've added new component but i couldn't manage to understand the part of "match.params.id"  to redirect the page to the single user details

File: src/PersonView.js [hunk 5]
```diff
@@ -1,31 +1,36 @@
...
-              <a href="https://jsonplaceholder.typicode.com/users/?id={person.id}">
-                Press Here
-              </a>
-            </ul>
-          </div>
-        )}
-      </li>
-    </div>
+    <BrowserRouter>
+      <div className="PersonView">
```
Comment (heshamelmasry77) [inline src/PersonView.js]: not correct , we will discuss this today together.

File: package.json [hunk 6]
```diff
@@ -10,7 +10,10 @@
     "node-sass": "^4.14.1",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
+<<<<<<< HEAD
     "react-loader-spinner": "^3.1.14",
+=======
```
Comment (heshamelmasry77) [inline package.json]: fix conflicts

File: src/PersonView.js [hunk 7]
```diff
@@ -1,12 +1,18 @@
 import React from "react";
+<<<<<<< HEAD
 import { BrowserRouter, Link } from "react-router-dom";
 //import UserDetail from "./UserDetail";
 import "./PersonView.scss";
+=======
+import {NavLink} from 'react-router-dom';
```
Comment (heshamelmasry77) [inline src/PersonView.js]: conflicts here too

File: src/PersonView.js [hunk 8]
```diff
@@ -1,12 +1,18 @@
...
+import './PersonView.scss';
+>>>>>>> 4055ad06d8afc25c442a505dd2c66fb9eb4d0dbe
 
 const PersonView = (props) => {
   const person = props.personData;
 
   return (
+<<<<<<< HEAD
```
Comment (heshamelmasry77) [inline src/PersonView.js]: conflicts

**Comment by heshamelmasry77** [comment]
too many conflicts :(, if you understand what is a conflict and how to fix it then try to fix it.

**Review by heshamelmasry77 — COMMENTED** [review]
did you make sure that this works ?

File: src/PersonView.js (hunk 8 above)
Comment (Mus90) [reply src/PersonView.js]: It is working, I will merge this branch with the second one

**Pull request closed without merging**